src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -229,18 +228,23 @@ impl<S: StateStore> StorageTableInner<S, EitherSerde> {
             .iter()
             .map(|idx| all_data_types[*idx].clone())
             .collect_vec();
-        let column_ids = value_indices
-            .iter()
-            .map(|idx| table_columns[*idx].column_id)
-            .collect_vec();
         let pk_serializer = OrderedRowSerde::new(pk_data_types, order_types);
 
         let row_serde = {
-            let schema = Arc::from(data_types.into_boxed_slice());
             if versioned {
-                ColumnAwareSerde::new(&column_ids, schema).into()
+                ColumnAwareSerde::new(
+                    Arc::from(value_indices.into_boxed_slice()),
+                    Arc::from(data_types.into_boxed_slice()),
+                    Arc::from(table_columns.into_boxed_slice()),
```

Comment:
The `data_types` here is already projected with `value_indices`, while `table_columns` is not, which seems confusing to me. I guess we can directly derive `data_types` from `table_columns`?